src/Poseidon/EntityTypes.hs

```diff
@@ -285,13 +306,13 @@ resolveUniqueEntityIndices entities indInfoCollection = do
     return relevantIndices
 
 -- | this returns a list of entities which could not be found
-determineNonExistentEntities :: (MonadThrow m, EntitySpec a) => [a] -> IndividualInfoCollection -> m EntitiesList
-determineNonExistentEntities entities indInfoCollection = do
-    return [ entity | entity <- map underlyingEntity entities, indices <- resolveEntityIndices [entity] indInfoCollection, null indices]
+determineNonExistentEntities :: (EntitySpec a) => [a] -> IndividualInfoCollection -> EntitiesList
+determineNonExistentEntities entities indInfoCollection =
+    [ entity | entity <- map underlyingEntity entities, null (resolveEntityIndices False [entity] indInfoCollection)]
```

Comment:
I think 

```haskell
[... | ..., indices <- resolveEntityIndices [entity] indInfoCollection, null indices ]
```

is more easy to read than the new

```haskell
[... | ..., null (resolveEntityIndices False [entity] indInfoCollection) ]
```

but as that's just a matter of taste.

Reply:
That doesn't work anymore, I think. At least it throws a compiler error. I am not sure I fully understand it, to be honest. I agree it looks more elegant, but even before I found this code a bit unintuitive. Usually such a double assignment `[... | a <- as, b <- bs]` is used for double-loops, but here it was used differently. Not sure. In any case, you're welcome to try and turn it back to how it was. I couldn't.

Reply:
I understand. `resolveEntityIndices` previously lived in MonadThrow. Now it's a pure function and this list comprehension now handles it as a normal producer. Ok.